src/components/Card/AddressHashCard.tsx

```diff
@@ -69,27 +70,39 @@ const AddressHashCardPanel = styled.div`
       }
     }
   }
-`
 
-const handleHashText = (hash: string) => {
-  if (isSmallMobile()) {
-    return startEndEllipsis(hash, 8, 10)
-  }
-  if (isMediumMobile()) {
-    return startEndEllipsis(hash, 9)
-  }
-  if (isLargeMobile()) {
-    return startEndEllipsis(hash, 14)
+  #address_hash__hash {
+    color: #ffffff;
   }
-  return hash
-}
+`
 
 export default ({ title, hash, dispatch }: { title: string; hash: string; dispatch: AppDispatch }) => {
+  const [hashText, setHashText] = useState(hash)
+  const isMobileDeivce = isMobile()
+
+  useLayoutEffect(() => {
+    if (!isMobile()) {
+      setHashText(hash)
+      return
+    }
+    const contentElement = document.getElementById('address_hash_content')
+    const hashElement = document.getElementById('address_hash__value')
+    if (hashElement && contentElement) {
+      const contentReact = contentElement.getBoundingClientRect()
+      const hashReact = hashElement.getBoundingClientRect()
+      const textWidth = contentReact.width - hashReact.left - 16 - 20
+      const textLength = Math.round(textWidth / 8.0)
+      const startLength = Math.round(textLength / 2)
+      const text = startEndEllipsis(hash, textLength - startLength, startLength)
+      setHashText(text)
+    }
+  }, [hash, isMobileDeivce])
+
   return (
-    <AddressHashCardPanel>
+    <AddressHashCardPanel id="address_hash_content">
       <div className="address_hash__title">{title}</div>
-      <div id="address_hash__hash">
-        <code>{handleHashText(hash)}</code>
+      <div id="address_hash__value">
+        <code>{hashText}</code>
       </div>
       <div
         className="address_hash__copy_iocn"
```

Comment:
icon not iocn

Reply:
I have modified the name. https://github.com/nervosnetwork/ckb-explorer-frontend/pull/168/commits/6ba488ca7dc7217a70d48c14a508b78f4bc9080a